### ISSUE TITLE
Error failed to find a model provider for model [gpt-4o]

### ISSUE THREAD

**Issue by kamalbuilds** [gptscript-ai/gptscript#584]
Error failed to find a model provider for model [gpt-4o]

**Comment by drpebcak** [comment]
Hi @kamalbuilds, can you provide a little bit more context here?

* What command are you running that outputs this error?
* What version of gptscript?
* what script are you trying to run?
* is there any other output?
* can you confirm that you have api access to gpt-4o?
* if you run `gptscript --list-models` do you see `gpt-4o` in the output?

Information like this will make it easier to troubleshoot your issue.

**Comment by kamalbuilds** [comment]
"@gptscript-ai/gptscript": "^0.8.3",
    
Running 
```
const run = await g.run(script, opts);
```
leads me to the errror , Below are the files and the output of console for reference

```
import { GPTScript } from "@gptscript-ai/gptscript";

const g = new GPTScript({
  APIKey: process.env.OPENAI_API_KEY,
  DefaultModel: "gpt-3.5-turbo"
});

export default g;
```

route.ts
```
import { NextRequest } from "next/server";
import { RunEventType, RunOpts } from "@gptscript-ai/gptscript";
import g from "@/lib/gptScriptInstance";

const script = "app/api/run-script/story-book.gpt";

export async function POST(request: NextRequest) {
  // to get the body we do
  const { story, pages, path } = await request.json();

  const opts: RunOpts = {
    disableCache: true,
    input: `--story ${story} --pages ${pages} --${path}`,
  };

  try {
    const encoder = new TextEncoder();
    const stream = new ReadableStream({
      async start(controller) {
        try {
          g.listModels().then((models) => {
            console.log("Models:", models);
          });
          console.log("Running gptscript with options:", opts);
          
          const run = await g.run(script, opts);
          console.log("GPTScript run initiated:", run);

          run.on(RunEventType.Event, (data) => {
            controller.enqueue(
              encoder.encode(`event: ${JSON.stringify(data)}\n\n`)
            );
          });
          await run.text();
          controller.close();
        } catch (error) {
          controller.error(error);
          console.error("Error", error);
        }
      },
    });
    return new Response(stream, {
      headers: {
        "Content-Type": "text/event-stream",
        "Cache-Control": "no-cache",
        Connection: "keep-alive",
      },
    });
  } catch (error) {
    console.error("Error in POST handler:", error);
    return new Response(JSON.stringify({ error: error }), {
      status: 500,
    });
  }
}

```
No I dont see `gpt-4o` in the output

**Comment by thedadams** [comment]
This is the same issue as you reported in https://github.com/gptscript-ai/gptscript/issues/583

This issue was fixed in `v0.8.5`. Please try upgrading. If the problem persists, please use the other issue to give more details.

Closing this issue.
